iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergMetaHook.java

```diff
@@ -262,17 +264,52 @@ public void preAlterTable(org.apache.hadoop.hive.metastore.api.Table hmsTable, E
 
     if (AlterTableType.ADDCOLS.equals(currentAlterTableOp)) {
       Collection<FieldSchema> addedCols =
-          HiveSchemaUtil.schemaDifference(hmsTable.getSd().getCols(), HiveSchemaUtil.convert(icebergTable.schema()));
+          HiveSchemaUtil.newColumns(hmsTable.getSd().getCols(), HiveSchemaUtil.convert(icebergTable.schema()));
+      if (!addedCols.isEmpty()) {
+        updateSchema = icebergTable.updateSchema();
+      }
+      for (FieldSchema addedCol : addedCols) {
+        updateSchema.addColumn(
+            addedCol.getName(),
+            HiveSchemaUtil.convert(TypeInfoUtils.getTypeInfoFromTypeString(addedCol.getType())),
+            addedCol.getComment()
+        );
+      }
+    } else if (AlterTableType.REPLACE_COLUMNS.equals(currentAlterTableOp)) {
+      List<FieldSchema> icebergSchema = HiveSchemaUtil.convert(icebergTable.schema());
+      List<FieldSchema> hmsSchema = hmsTable.getSd().getCols();
+      Collection<FieldSchema> droppedCols = HiveSchemaUtil.newColumns(icebergSchema, hmsSchema);
```

Comment:
I understand why `newColums` are used, but maybe a better name could be found instead of `newColumns`, like `missingColumns`?

Reply:
See below. refactored the code